component/module/http/http.go

```diff
@@ -0,0 +1,188 @@
+}
+
+// Component implements the module.file component.
+type Component struct {
+	opts component.Options
+	mod  *module.ModuleComponent
+
+	mut     sync.RWMutex
+	args    Arguments
+	content rivertypes.OptionalSecret
+
+	managedRemoteHTTP *remote_http.Component
+	inUpdate          atomic.Bool
+	isCreated         atomic.Bool
+}
+
+var (
+	_ component.Component       = (*Component)(nil)
+	_ component.HealthComponent = (*Component)(nil)
+	_ component.HTTPComponent   = (*Component)(nil)
+)
+
+// New creates a new module.file component.
+func New(o component.Options, args Arguments) (*Component, error) {
+	m, err := module.NewModuleComponent(o)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &Component{
+		opts: o,
+		mod:  m,
+		args: args,
+	}
+	defer c.isCreated.Store(true)
+
+	c.managedRemoteHTTP, err = c.newManagedLocalComponent(o)
+	if err != nil {
+		return nil, err
+	}
+	if err := c.Update(args); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// NewManagedLocalComponent creates the new remote.http managed component.
+func (c *Component) newManagedLocalComponent(o component.Options) (*remote_http.Component, error) {
+	localFileOpts := o
+	localFileOpts.OnStateChange = func(e component.Exports) {
+		c.setContent(e.(remote_http.Exports).Content)
+
+		if !c.inUpdate.Load() && c.isCreated.Load() {
+			// Any errors found here are reported via component health
+			_ = c.mod.LoadFlowContent(c.getArgs().Arguments, c.getContent().Value)
+		}
+	}
+
+	return remote_http.New(localFileOpts, c.getArgs().RemoteHTTPArguments)
+}
+
+// Run implements component.Component.
+func (c *Component) Run(ctx context.Context) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	ch := make(chan error, 1)
+	go func() {
+		err := c.managedRemoteHTTP.Run(ctx)
+		if err != nil {
+			ch <- err
+		}
+	}()
+
+	go c.mod.RunFlowController(ctx)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case err := <-ch:
+			return err
+		}
+	}
+}
+
+// Update implements component.Component.
+func (c *Component) Update(args component.Arguments) error {
+	c.inUpdate.Store(true)
+	defer c.inUpdate.Store(false)
+
+	newArgs := args.(Arguments)
+	c.setArgs(newArgs)
+
+	err := c.managedRemoteHTTP.Update(newArgs.RemoteHTTPArguments)
+	if err != nil {
+		return err
+	}
+
+	// Force a content load here and bubble up any error. This will catch problems
+	// on initial load.
+	return c.mod.LoadFlowContent(newArgs.Arguments, c.getContent().Value)
+}
+
+// Handler implements component.HTTPComponent.
+func (c *Component) Handler() http.Handler {
+	return c.mod.HTTPHandler()
+}
+
+// CurrentHealth implements component.HealthComponent.
+func (c *Component) CurrentHealth() component.Health {
+	// Note that it takes until the first successful poll for c.managedRemoteHTTP to
+	// become healthy.
+	leastHealthy := component.LeastHealthy(
+		c.managedRemoteHTTP.CurrentHealth(),
+		c.mod.CurrentHealth(),
+	)
+
+	if leastHealthy.Health == component.HealthTypeHealthy {
```

Comment:
This looks like a copy of module.file, and part of me thinks we added it to fix a bug but still not sure why its not extraneous code.

Reply:
Yeah, this is heavily brought over from `module.file` since they are mostly functionally the same, just with a different source (local file vs. body of remote http call).